_posts/2018-08-29-generate-and-install-a-root-certificate.adoc

```diff
@@ -0,0 +1,80 @@
+= Generate and install root certificates
+:page-layout: page
+:page-categories: [prod]
+:page-date: 2018-08-29 11:12:06
+:page-order: 2
+:icons: font
+
+Before you can get started with implicit provisioning, you'll need the following components:
+
+- A root CA certificate and private key
+- Device credentials that are signed by root CA certificate
+- The internal root CA certificate of your device gateway
+
+The following instructions describe how you can generate these components. Getting a root CA certificate is beyond the scope of these instructions, but we can show you how to generate a self-signed certificate for testing.
+// tag::install-root-ca[]
+
+To generate and install root certificates, follow these steps: ::
+. Generate a root CA private key and self-signed certificate.
++
+If you do not have your own CA certificate for signing device certificates, you can generate a self-signed certificate for testing.
++
+ifdef::env-github[]
+You can examine the `new_server` function in link:https://github.com/advancedtelematic/ota-community-edition/blob/master/scripts/start.sh#L127[OTA Community Edition's `start.sh`] for one way of generating the cert.
+endif::[]
```

Comment:
All of the `ifdef`/`ifndef` stuff should be removed; that's only relevant if we're including it from the aktualizr repo. Since you've brought it in here, we can be a lot clearer.

(also, if we aren't doing the `include::[]` from the aktualizr repo anymore, we might as well clean up the version in the aktualizr repo as well--it wouldn't need the ifdef/ifndef blocks either.)

Reply:
OK fixed and also created a pull request in the aktualizr repo: https://github.com/advancedtelematic/aktualizr/pull/935